games/armagetronad/Portfile

```diff
@@ -0,0 +1,291 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+PortGroup           boost 1.0
+
+github.setup        ArmagetronAd armagetronad 0.2.9.1.0 v
+github.tarball_from releases
+extract.suffix      .tbz
+
+set pretty_name     "Armagetron Advanced"
+# 'year_released' is the year that the tarball was published
+# on GitHub, not the year that we are currently in.
+set year_released   2021
+
+categories          games
+license             GPL-2+
+maintainers         @jasonliu-- openmaintainer
+                    #  v Note: https site has an expired certificate
+homepage            http://www.${name}.org
+description         $pretty_name, a 3-D Tron-like light cycle game
+long_description    $pretty_name is a 3-D snake game, based on the \
+                    light cycle sequence from the movie Tron. You can \
+                    play against AI opponents, against other human \
+                    users over the network, or a mixture of both.
+
+subport ${name}-common {
+    description     common files for the $pretty_name packages
+    long_description-append \
+                    \n\nThis package contains files that are common to \
+                    both the ${name} and ${name}-dedicated packages.
+}
+subport ${name}-dedicated {
+    description     dedicated game server for $pretty_name
+    long_description-append \
+                    \n\nThis package installs the dedicated network \
+                    server for the $pretty_name game. It allows you to \
+                    host games that other online players can connect \
+                    to, and play multi-player matches. You can also \
+                    host online multi-player matches directly from \
+                    inside the game, but the dedicated server allows \
+                    you to continue hosting matches while not actively \
+                    running the game.
+}
+
+distfiles           ${distname}${extract.suffix}:${name}
+checksums           ${distname}${extract.suffix} \
+                    rmd160  76eed9dd9fe6d888eba0fae67696040ec1eccb7d \
+                    sha256  59b6c7c01ce3f8cca5437e33f974a637529541a11aa4f52c1a5c17499e26f6a1 \
+                    size    1943758
+
+if {$subport ne "${name}-common"} {
+    master_sites-append http://deb.debian.org/debian/pool/main/a/${name}:debian_pkg
+    distfiles-append    ${name}_${version}-2.debian.tar.xz:debian_pkg
```

Comment:
Ah, and this is an xz-compressed tarball? MacPorts does not support automatically decompressing multiple distfiles of different types. (I haven't gotten around to adding this capability yet, though I would like to.) You should let MacPorts extract one of the files (perhaps the main .tbz file), list that file in `extract.only`, and extract any additional distfiles with differing compression formats manually in a `post-extract` block with `system`. Don't forget to add `depends_extract bin:xz:xz` if you need to extract xz files.

Reply:
> MacPorts does not support automatically decompressing multiple distfiles of different types. (I haven't gotten around to adding this capability yet, though I would like to.)

Interesting... it seems to work fine for this Portfile, at least when I tested it on macOS 10.11. I suspect it might have something to do with what you mention in your above comment, about the "wrong" extraction command happening to work on OS X 10.9 and later. Does that mean that I only need to add that stuff for the sake of macOS <= 10.9? Because adding an `extract.only`, `post-extract`, `depends_extract`, etc. seems like a lot of extra work just for the sake of extracting a couple of distfiles, and obviously I wouldn't have submitted the PR if the entire Portfile hadn't worked, and it does seem to work flawlessly, even with `port -vst` turned on.

Reply:
After thinking it over, how about this? Since the current form of the portfile seems to "just work", both locally on my 10.11 machine, as well as the 10.15 and 11.0 CI builds, why don't we merge the portfile as-is, and then I can see which macOS versions might be having a problem by looking at the buildbot logs, and tailor a PR for those (probably older) macOSes?

Reply:
FYI, I have used this to extract multiple different distfiles of different types in my own ports for some years:
```
extract {
    foreach f ${distfiles} {
        set f [lindex [split ${f} :] 0]
        ui_msg "Extracting: ${f}"
        if {[string match "*bz2*" ${f}]} {
            ui_msg "${workpath} \"/usr/bin/tar xjf ${distpath}/${f}\""
            system -W ${workpath} "/usr/bin/tar xjf ${distpath}/${f}"
        }
        if {[string match "*gz*" ${f}]}  {
            ui_msg "${workpath} \"/usr/bin/tar xzf ${distpath}/${f}\""
            system -W ${workpath} "/usr/bin/tar xzf ${distpath}/${f}"
        }
    }
}
```

add types as needed

Reply:
Yes, the problem only affects OS X < 10.9.